stock_storage_type/models/stock_location.py

```diff
@@ -379,10 +383,55 @@ def get_storage_locations(self, products=None):
         locations = self.browse()
         if self.pack_putaway_strategy == "none":
             locations = self
-        elif self.pack_putaway_strategy == "ordered_locations":
-            locations = self._get_ordered_leaf_locations()
+        else:
+            locations = self._get_sorted_leaf_locations(products)
```

Comment:
There is a fix for this here https://github.com/OCA/wms/pull/280/files

Reply:
Fix cherry-picked